internal/cli/cli.go

```diff
@@ -38,83 +42,180 @@ type CommandDefinition struct {
 }
 
 // CommandFactory is a command factory
-type CommandFactory struct {
-	Profile *Profile
-	Config  *Config
+type CommandFactory interface {
+	Build(func() CommandDefinition) *cobra.Command
+	Close()
+	Init()
+	Run(cmd *cobra.Command)
+	SetGlobalFlags(fs *flag.FlagSet)
+}
+
+type commandFactory struct {
+	in      *os.File
+	out     *os.File
+	err     *os.File
+	config  *Config
+	profile *Profile
+	ui      terminal.UI
+	logger  *log.Logger
 }
 
 // Config is the global CLI config
 type Config struct {
-	Command CommandConfig
-	UI      terminal.UIConfig
+	CommandConfig
+	terminal.UIConfig
 }
 
 // CommandConfig holds the global config for a CLI command
 type CommandConfig struct {
 	RealmBaseURL string
 }
 
+// NewCommandFactory creates a new command factory
+func NewCommandFactory() CommandFactory {
+	logger := log.New(os.Stderr, "UTC ERROR ", log.Ltime|log.Lmsgprefix)
+
+	config := new(Config)
+
+	profile, profileErr := NewDefaultProfile()
+	if profileErr != nil {
+		logger.Fatal(profileErr)
+	}
+
+	return &commandFactory{
+		config:  config,
+		profile: profile,
+		logger:  logger,
+	}
+}
+
+func (factory *commandFactory) Init() {
+	if err := factory.profile.Load(); err != nil {
+		factory.logger.Fatal(err)
+	}
+
+	if filepath := factory.config.OutputTarget.Path(); filepath != "" {
```

Comment:
[super nit] I always try to un-nest (de-nest?) as much as possible. Thoughts on the inverse here?
```
filepath := factory.config.OutputTarget.Path()
if filepath == "" {
    return
}
.
.
```

Reply:
totally fine to switch to it if you feel strongly about that, but I don't love `return` here since it makes the ordering of these `Setup` steps important (all new `Setup` would need to go be between this block and `profile.Load()` above)